src/FrameStatisticsDialog.cpp

```diff
@@ -0,0 +1,149 @@
+         {
+            S.AddFixedText(Verbatim("Track Panel Rendering"));
+            AddSection(S, FrameStatistics::SectionID::TrackPanel);
+            S.AddFixedText(Verbatim("Waveform Rendering (per clip)"));
+            AddSection(S, FrameStatistics::SectionID::WaveformView);
+            S.AddFixedText(Verbatim("Spectrum Rendering (per clip)"));
+            AddSection(S, FrameStatistics::SectionID::SpectrumView);
+         }
+         S.EndVerticalLay();
+      }
+      S.EndPanel();
+
+      Layout();
+      Fit();
+
+      mStatisticsUpdated = FrameStatistics::Subscribe(
+         [this](FrameStatistics::SectionID sectionID) {
+            constexpr auto timeout = std::chrono::milliseconds(300);
+            const auto now = std::chrono::steady_clock::now();
+
+            if ((now - mLastUpdate) > timeout)
+            {
+               SectionUpdated(sectionID);
+               mLastUpdate = now;
+            }
+         });
+   }
+
+private:
+   void AddSection(ShuttleGui& S, FrameStatistics::SectionID sectionID)
+   {
+      S.StartMultiColumn(2, wxEXPAND);
+      {
+         S.AddFixedText(Verbatim("Last:"));
+         mSections[size_t(sectionID)].Last = S.AddVariableText({});
+
+         S.AddFixedText(Verbatim("Min:"));
+         mSections[size_t(sectionID)].Min = S.AddVariableText({});
+
+         S.AddFixedText(Verbatim("Max:"));
+         mSections[size_t(sectionID)].Max = S.AddVariableText({});
+
+         S.AddFixedText(Verbatim("Avg:"));
+         mSections[size_t(sectionID)].Avg = S.AddVariableText({});
+
+         S.AddFixedText(Verbatim("Events:"));
+         mSections[size_t(sectionID)].Events = S.AddVariableText({});
+      }
+      S.EndMultiColumn();
+
+      SectionUpdated(sectionID);
+   }
+
+   wxString FormatTime (FrameStatistics::Duration duration)
+   {
+      using namespace std::chrono;
+
+      const auto mcs = duration_cast<microseconds>(duration);
+
+      return std::to_string(mcs.count() / 1000.0) + " ms";
+   }
+
+   void SectionUpdated(FrameStatistics::SectionID sectionID)
+   {
+      Section& section = mSections[size_t(sectionID)];
+      const auto& profilerSection = FrameStatistics::GetSection(sectionID);
+
+      section.Last->SetLabel(FormatTime(profilerSection.GetLastDuration()));
+      section.Min->SetLabel(FormatTime(profilerSection.GetMinDuration()));
+      section.Max->SetLabel(FormatTime(profilerSection.GetMaxDuration()));
+      section.Avg->SetLabel(FormatTime(profilerSection.GetAverageDuration()));
+      section.Events->SetLabel(std::to_string(profilerSection.GetEventsCount()));
+   }
+
+   struct Section final
+   {
+      wxStaticText* Last;
+      wxStaticText* Min;
+      wxStaticText* Max;
+      wxStaticText* Avg;
+      wxStaticText* Events;
+   };
+
+   Section mSections[size_t(FrameStatistics::SectionID::Count)];
+
+   Observer::Subscription mStatisticsUpdated;
+
+   std::chrono::steady_clock::time_point mLastUpdate;
+};
+
+Destroy_ptr<Dialog> sDialog;
```

Comment:
I wish to deprecate Destroy_ptr in favor of: https://docs.wxwidgets.org/3.0/classwx_window_ptr_3_01_t_01_4.html

I didn't know of that when I reinvented Destroy_ptr.